pkg/insights/insightsclient/insightsclient.go

```diff
@@ -40,6 +43,7 @@ type Client struct {
 
 type Authorizer interface {
 	Authorize(req *http.Request) error
+	HTTPConfig() config.HTTPConfig
```

Comment:
Why is the authorizer not handling the proxy setup?  It already has all the config info.

Reply:
I moved it there. If you ment something like a Roundtripper with dynamic injectable Proxy, I can prepare it as well, but at first it seemed a bit too much code.